chain/boltdb/store_test.go

```diff
@@ -149,3 +153,69 @@ func TestStoreBolt(t *testing.T) {
 	_, err = store.Get(ctx, 10000)
 	require.Equal(t, chainerrors.ErrNoBeaconStored, err)
 }
+
+func TestStore_Cursor(t *testing.T) {
+	tmp := t.TempDir()
+	ctx := context.Background()
+	l := test.Logger(t)
+	dbStore, err := NewBoltStore(l, tmp, nil)
+	require.NoError(t, err)
+	defer func() {
+		require.NoError(t, dbStore.Close(ctx))
+	}()
+
+	sigs := map[int][]byte{
+		0: {0x01, 0x02, 0x03},
+		1: {0x02, 0x03, 0x04},
+		2: {0x03, 0x04, 0x05},
+		3: {0x04, 0x05, 0x06},
+		4: {0x05, 0x06, 0x07},
+		5: {0x06, 0x07, 0x08},
+	}
+
+	beacons := make(map[int]*chain.Beacon, len(sigs)-1)
+
+	for i := 1; i < len(sigs); i++ {
+		b := &chain.Beacon{
+			PreviousSig: sigs[i-1],
+			Round:       uint64(i),
+			Signature:   sigs[i],
+		}
+
+		beacons[i] = b
```

Comment:
won't this be out of bounds on the last iteration if `len(beacons)` is `len(sigs) - 1`?

Reply:
No, because `len(signs)-1` is `5` and the map uses a 0-based index, and ends at 5.

So, the loop goes from 1 to 5, which is a valid range.

When we construct the map, we hint that we'll store `len(signs)-1` elements, but the number of elements is not related to the index values.

Reply:
ahhhh I'm an idiot - I thought it was an array x) I see now it's a map indeed